src/System.Buffers/src/System/Buffers/ArrayPoolBucket.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Provides a thread-safe bucket containing buffers that can be Rented and Returned as part 
+    /// of a buffer pool; it should not be used independent of the pool.
+    /// </summary>
+    internal sealed class ArrayPoolBucket<T>
+    {
+        private volatile int _index;
+        private T[][] _data;
```

Comment:
This is written to in Debug more so help track usage, near the end of the Rent call